Sources/Partial/Codable/KeyPathCodingKeyCollection.swift

```diff
@@ -0,0 +1,34 @@
+public struct KeyPathCodingKeyCollection<Root, CodingKey: Swift.CodingKey & Hashable> {
+    private typealias Encoder = (_ value: Any, _ container: inout KeyedEncodingContainer<CodingKey>) throws -> Void
+    private typealias Decoder = (_ codingKey: CodingKey, _ container: KeyedDecodingContainer<CodingKey>) throws -> (Any, PartialKeyPath<Root>)?
+
+    private var encoders: [PartialKeyPath<Root>: Encoder] = [:]
+    private var decoders: [CodingKey: Decoder] = [:]
+
+    func encode(_ value: Any, forKey keyPath: PartialKeyPath<Root>, to container: inout KeyedEncodingContainer<CodingKey>) throws {
+        try encoders[keyPath]?(value, &container)
```

Comment:
I hadn't considered that. It may be useful for consumers to be able to opt out certain values from being encoded, which making this throw would prevent.

But having this throw would also make debugging easier if a key path/coding key is missed.

Would it make sense to allow the `CodingKey` returned be `nil`, and if it's `nil` it'll never be encoded, but if it's missing it'll throw here?